header-rewrite-filter/header_processor.cc

```diff
@@ -212,11 +212,13 @@ namespace HeaderRewriteFilter {
     }
 
     absl::Status SetBoolProcessor::parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start) {
-        if (operation_expression.size() < Utility::SET_BOOL_MIN_NUM_ARGUMENTS) {
-            return absl::InvalidArgumentError("not enough arguments for set-bool");
-        }
-
         try {
+            if (operation_expression.size() < Utility::SET_BOOL_MIN_NUM_ARGUMENTS) {
+                return absl::InvalidArgumentError("not enough arguments for set-bool");
+            }
+
+            const bool isRequest = (operation_expression.at(0) == "http-request");
```

Comment:
When you create a SetBoolProcessor in you've already checked if it's a request [here](https://github.com/DataDog/envoy-filter-example/blob/main/header-rewrite-filter/header_rewrite.cc#L53), so I think we could just store the result as a member variable

Reply:
Isn't this state information specific to the HTTP request/response? This goes back to #27 where I removed any HTTP-specific state to keep the processors thread safe.

Reply:
The bool processor is tied to a request or response is based on the config e.g. the bool processor created from
```
http-request set-bool true_bool hdr(mock_header,-1) -m str mock_val
```
is always tied to a request, it's not info specific to a particular request or response

Reply:
Made this change in #34.